src/DssDirectDepositHub.sol

```diff
@@ -314,6 +314,8 @@ contract DssDirectDepositHub {
 
             // Determine if it needs to unwind due plan
             uint256 targetAssets = ilks[ilk_].plan.getTargetAssets(currentAssets);
+            // Determine if the pool limits our total assets
+            targetAssets = _min(targetAssets, pool.maxDeposit());
```

Comment:
I still don't understand well the use of `pool.maxDeposit()` but if it is directly overwritting `targetAssets` can't this be implemented in `plan.getTargetAssets(currentAssets)`?

Reply:
Yes I thought of that... However, we have said two things that suggested it shouldn't be there:
1 - We have decided `plan`s and `pool`s should not be connected. This would entangle them if we have the `maxDeposit` in the pool and call it from the plan. We could always have `maxDeposit` in the plan, but that feels weird to have `maxWithdraw` in the pool and `maxDeposit` in the plan.
2 - We have said that plans could be used to service multiple D3Ms/Pools.  This feels like it could make `maxDeposit` more complicated/confusing to have in the plans.

If we do move it to the plan though we could exclude it from the interface and hub which would be nice

Reply:
I guessed it was related to that separation, so I think is actually fine as it is.